sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthenticationUtil.java

```diff
@@ -0,0 +1,90 @@
+     * HttpClient client = HttpClient.createDefault&#40;&#41;;
+     * client.sendSync&#40;request, Context.NONE&#41;;
+     * </pre>
+     * <!-- end com.azure.identity.util.getBearerTokenSupplierSync -->
+     *
+     * @param credential The {@link TokenCredential} from which to retrieve a token.
+     * @param scopes The scopes as appropriate for the token you are retrieving.
+     * @return A {@link Supplier<String>} which returns the bearer token.
+     */
+    public static Supplier<String> getBearerTokenSupplierSync(TokenCredential credential, String... scopes) {
+        HttpPipeline pipeline = new HttpPipelineBuilder().policies(new BearerTokenAuthenticationPolicy(credential, scopes)).build();
+        return () -> {
+            // This request will never need to go anywhere; it is simply to cause the policy to interact with
+            // the user's credential
+            HttpRequest req = new HttpRequest(HttpMethod.GET, "https://www.example.com");
+            try (HttpResponse res = pipeline.sendSync(req, Context.NONE)) {
+                return res.getRequest().getHeaders().get(HttpHeaderName.AUTHORIZATION).getValue().split(" ")[1];
+            }
+        };
+    }
+
+    /**
+     * Creates a {@link Supplier<Mono<String>>} that provides a Bearer token from the specified credential.
+     * The token is cached and will refresh when it expires.
+     * <p><strong>Using the supplier:</strong></p>
+     * <!-- src_embed com.azure.identity.util.getBearerTokenSupplier -->
+     * <pre>
+     * DefaultAzureCredential credential = new DefaultAzureCredentialBuilder&#40;&#41;.build&#40;&#41;;
+     * String scope = &quot;https:&#47;&#47;cognitiveservices.azure.com&#47;.default&quot;;
+     * Supplier&lt;Mono&lt;String&gt;&gt; supplier = AuthenticationUtil.getBearerTokenSupplier&#40;credential, scope&#41;;
+     *
+     * &#47;&#47; This example simply uses the Azure SDK HTTP library to demonstrate setting the header.
+     * &#47;&#47; Use the token as is appropriate for your circumstances.
+     * HttpRequest request = new HttpRequest&#40;HttpMethod.GET, &quot;https:&#47;&#47;www.example.com&quot;&#41;;
+     * supplier.get&#40;&#41;.doOnNext&#40;token -&gt;
+     * &#123;
+     *     request.setHeader&#40;HttpHeaderName.AUTHORIZATION, &quot;Bearer &quot; + token&#41;;
+     *     HttpClient client = HttpClient.createDefault&#40;&#41;;
+     *     client.send&#40;request, Context.NONE&#41;;
+     * &#125;&#41;.subscribe&#40;&#41;;
+     * </pre>
+     * <!-- end com.azure.identity.util.getBearerTokenSupplier -->
+     *
+     * @param credential The {@link TokenCredential} from which to retrieve a token.
+     * @param scopes The scopes as appropriate for the token you are retrieving.
+     * @return A {@link Supplier<Mono<String>>} which returns the bearer token.
+     */
+    public static Supplier<Mono<String>> getBearerTokenSupplier(TokenCredential credential, String... scopes) {
```

Comment:
For any new APIs, we should add `Async` suffix for the async methods rather than `Sync` suffix to sync methods. 

Also, do we need the async method? I don't see a need for this in OpenAI as we don't want to bring Reactor APIs into that library if we can avoid it. So, let's start with only the sync method for now and add the async method later, if required.

Reply:
Sounds good - @alzimmermsft had the reasonable point here but we can always come back.